KAS-3567: Kort Bestek user profile

sergiofenoll opened the pull request:
https://github.com/kanselarij-vlaanderen/app-kaleidos/pull/306

Jenkins run: http://kal-kastaar.s.redpencil.io:8080/job/kaleidos/job/custom_frontend_backend_combination/267/

sergiofenoll committed:
feat: set /kort-bestek as default landing for Kort Bestek user profile

sergiofenoll committed:
feat: set kort bestek as default search subtab for Kort Bestek user profile

erikap commented:
Nog niet in depth bekeken, maar op het eerste zicht: kunnen mensen van Kort Bestek de nieuwsberichten nu nog editeren aangezien `currentSession.isEditor` niet meer geldt voor hen? 
Ik had n.a.v. van deze PR eerder wat changes verwacht die `isEditor` vervangen met nieuwe user roles om een duidelijker onderscheid te hebben tussen secretarie en kort bestek.

sergiofenoll committed:
feat: copy over user-may helper

From: https://github.com/kanselarij-vlaanderen/frontend-kaleidos/commit/c54ca7c3ae8bfa31d29d085d398410b95352b465

sergiofenoll committed:
feat: use manage-newsletter-infos role in Newsletter::NewsletterHeaderOverview

sergiofenoll committed:
refactor: use user-may helper in NewsletterItem::TableRow component

sergiofenoll committed:
feat: use group role manage-newsletter-infos in agendaitem route

sergiofenoll committed:
feat: remove redirect to kort bestek search tab

This will no longer be necessary once #1433 gets merged.

sergiofenoll commented:
> Nog niet in depth bekeken, maar op het eerste zicht: kunnen mensen van Kort Bestek de nieuwsberichten nu nog editeren aangezien `currentSession.isEditor` niet meer geldt voor hen? Ik had n.a.v. van deze PR eerder wat changes verwacht die `isEditor` vervangen met nieuwe user roles om een duidelijker onderscheid te hebben tussen secretarie en kort bestek.

Thanks voor de opmerking, ik ben een beetje te snel door de feature gegaan en heb er niet bij stilgestaan wat de gebruikers nu eenmaal gaan doen met dat profiel.

Kort Bestek gebruikers kunnen nu in de `newsletters.newsletter.index` en `agenda.agendaitems.agendaitem.news- item` routes editeren. Zijn er verdere acties die ze moeten kunnen uitvoeren?

sergiofenoll committed:
feat: use manage-settings role for access to settings routes

sergiofenoll committed:
feat: make Kort Bestek group an editor

This way Kort Bestek users will have the same permissions as before,
this is to prevent them lacking features that they could and still need
to use.

sergiofenoll committed:
Merge branch 'development' into feature/KAS-3567-kort-bestek-user-profile

erikap committed:
Merge branch 'development' into feature/KAS-3567-kort-bestek-user-profile

sergiofenoll committed:
Merge branch 'development' into feature/KAS-3567-kort-bestek-user-profile

erikap committed:
Add automatic transition to publications route for OVRB

erikap committed:
Move business logic about visibility of agendaitem tabs to controller

erikap committed:
Merge branch 'feature/KAS-3567-kort-bestek-user-profile' of github.com:kanselarij-vlaanderen/frontend-kaleidos into feature/KAS-3567-kort-bestek-user-profile

erikap committed:
Remove manage-settings role from kanselarij group

Only admins are allowed to manage settings

erikap committed:
Add manage-themis-publications role to kort bestek group

erikap committed:
Add manage-agenda-versions role to admin and kanselarij group

erikap committed:
Add manage-agendaitems and manage-meeting roles to admin and
kanselarij group

erikap commented:
Custom Jenkins run: http://kal-kastaar.s.redpencil.io:8080/job/kaleidos/job/custom_frontend_backend_combination/272

sergiofenoll commented on app/config/roles.js:
```diff
@@ -39,8 +39,7 @@ groupRoles.set(KANSELARIJ, [
   'manage-document-publications',
-  'manage-themis-publications',
-  'manage-settings',
+  'manage-themis-publications'
```
At least in the test we seem to expect Kanselarij group to be able to access the settings. If this isn't actually the case, I'll try to rework the tests that depend on that.

erikap replied:
Sorry, will re-add that one. Out of scope of this story :blush:

sergiofenoll committed:
fix: correct AgendaVersionActions modal conditions

erikap committed:
Add manage-settings back to kanselarij group

erikap commented:
Custom Jenkins run: http://kal-kastaar.s.redpencil.io:8080/job/kaleidos/job/custom_frontend_backend_combination/274
